### PR TITLE
AUE 388

### PR DESCRIPTION
**Pull request by jweible**
Example output:

Analysis Project: d98503f8cc9e49c1a3c9b6a30c94328d
Name: CLE Validation AML Pairs genome-3551
Status: Hold
Run as: cle-staging
Created At: 2014-12-07 13:49:39
Updated: 2014-12-18 10:29:37
Created By: jwalker

Subject Mapping: 0939546b37944677b6f2ec548aff557d
    Subject: 05eda9363b4c40599375b2a8fa476462
        Subject: 2811093070
        Label: tumor_sample
        Created By: jwalker
        Created At: 2014-12-07 13:59:35
        Updated At: 2014-12-07 13:59:35
        Tags: discovery

    Subject: 4db5235abaf34649b2a14876e21e6ed5...

### CODE REVIEW

File: lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/View.pm [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        ['Name', $analysis_project->name],
+        ['Status', $analysis_project->status],
+        ['Run as', $analysis_project->run_as],
+        ['Created At', $analysis_project->created_at],
+        ['Updated', $analysis_project->updated_at],
+        ['Created By', $analysis_project->created_by],
+    );
+}
+
+sub _write_pairs_line {
```
Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/View.pm]: It looks like `_write_pairs_line` was copied verbatim from the AnalysisProject view command.  Could this be pulled up into a parent class?

**Comment by tmooney** [comment]
It might be good to add a small test that sets up a test AnP and some subject mappings to make sure the command runs to completion.

**Comment by acoffman** [comment]
Personal preference, but I believe that most people specify their subject mappings on import based on name, so I wonder if it would be good to show the name here? That would slow the command down a bit, especially for large AnPs so maybe it should be an option? Or maybe a UR hint would help? I'm not sure exactly how that works...

**Comment by jasonwalker80** [comment]
Names would be much more beneficial for the end user over the UUID.

File: lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/View.pm [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+sub _get_subject_lines {
+    my ($self, $subject) = @_;
+    return (
+        ['Subject', $subject->subject->id ],
+        ['Label', $subject->label],
+        ['Created By', $subject->created_by], 
+        ['Created At', $subject->created_at], 
+        ['Updated At', $subject->updated_at], 
+        ['Tags', join(',', map{$_->name} $subject->subject_mapping->tags)],
```
Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/View.pm]: If the tags are properties of the mapping, they needn't be repeated with each subject in the mapping.

**Comment by jweible** [comment]
Commits added to address feedback.  Current output looks like so:

**Comment by jweible** [comment]
Analysis Project: d98503f8cc9e49c1a3c9b6a30c94328d
Name: CLE Validation AML Pairs genome-3551
Status: Hold
Run as: cle-staging
Created At: 2014-12-07 13:49:39
Updated: 2014-12-18 10:29:37
Created By: jwalker

Subject Mapping: 0939546b37944677b6f2ec548aff557d
Tags: discovery
    Subject: H_KA-702808-0909968
        Subject: 2811093070
        Label: tumor_sample
        Created By: jwalker
        Created At: 2014-12-07 13:59:35
        Updated At: 2014-12-07 13:59:35

    Subject: H_KA-702808-0900124
        Subject: 2811097611
        Label: normal_sample
        Created By: jwalker
        Created At: 2014-12-07 13:59:35
        Updated At: 2014-12-07 13:59:35

Subject Mapping: 1187e394a5a246c6b70629788a3343ea
Tags: germline 
...

**Comment by jweible** [comment]
Example output (with names):
Analysis Project: d98503f8cc9e49c1a3c9b6a30c94328d
Name: CLE Validation AML Pairs genome-3551
Status: Hold
Run as: cle-staging                      Created By: jwalker
Created: 2014-12-07 13:49:39             Updated: 2014-12-18 10:29:37

=== Subject Mapping (0939546b37944677b6f2ec548aff557d): ===
    Tags: discovery
    Subject: H_KA-702808-0909968 (AML37 tumor 2811093070)
        Label: tumor_sample
        Created By: jwalker
        Created: 2014-12-07 13:59:35             Updated: 2014-12-07 13:59:35

    Subject: H_KA-702808-0900124 (AML37 normal 2811097611)
        Label: normal_sample
        Created By: jwalker
        Created: 2014-12-07 13:59:35             Updated: 2014-12-07 13:59:35

=== Subject Mapping (1187e394a5a246c6b70629788a3343ea): ===
    Tags: germline
    Subject: H_KA-804168-0802136 (AML40 normal 2811092681)
        Label: tumor_sample
        Created By: jwalker
        Created: 2014-12-07 13:59:35             Updated: 2014-12-07 13:59:35

**Comment by jweible** [comment]
@tmooney : Duplicate Code removed.  A test that runs use_ok has been added.  I do not believe a test that just runs the command without checking the output is worthwhile, and getting this out has already taken too much time.

**Comment by jweible** [comment]
@acoffman & @jasonwalker80: Names are in.  See above for the latest example output

**Comment by jweible** [comment]
@tmooney: Acknowledged and fixed

**Comment by jasonwalker80** [comment]
+1

**Comment by acoffman** [comment]
+1

**Comment by tmooney** [comment]
+1 :facepunch:

**Pull request merged**